rad/monadic/items.rad

```diff
@@ -20,3 +20,178 @@
+         (show n)
+         tit
+         un
+         (pretty-modified-at i)]))))
+
+(def pretty-row
+  (fn [i]
+    (match i
+      '{:number n
+        :title tit
+        :state s
+        :username un}
+      (string-append
+        (wrap-cell (pretty-state s) :state)
+        (wrap-cell (show n) :number)
+        (wrap-cell tit :title)
+        (wrap-cell un :author)
+        (wrap-cell (pretty-modified-at i) :modified-at)))))
+
+(def pretty-comment
+ (fn [c]
+   (match c
+     '{:created-at c-a
+       :body b
+       :username un}
+     (string-append "### " un " [" c-a "]\n\n" b "\n"))))
+
+(def pretty-item-view
+  "Markdown view of an item."
+  (fn [i]
+    (def pretty-comments
+      (fn [comments]
+        (string/unlines (map pretty-comment comments))))
+    (match i
+      '{:number n
+        :title tit
+        :body b
+        :state state
+        :labels ls
+        :comments cs
+        :username un
+        :created-at c-a}
+      (string-replace
+       "\r"
+       ""
+       (string-append
+        (pretty-headline i) "\n\n"
+        "**State:** " (pretty-state state) "\n"
+        "**Labels:** " (show ls) "\n\n"
+        b "\n\n"
+        "Comments\n"
+        "--------\n\n"
+        (pretty-comments cs)))
+      '{:patch p
+        :created-at c-a
+        :comments cs}
+      (string-append
+        (pretty-headline i) "\n\n"
+        p "\n\n"
+        "Comments\n"
+        "--------\n\n"
+        (pretty-comments cs)))))
+
+(def items-name
+  (fn [item-flag]
+    (match item-flag
+      :issue  "issues"
+      :patch  "patches")))
+
+(def list-items
+  "Lists the recent items and uses an editor to add a relevant comment."
+  (fn [machine options items item-flag enrich-item add-comment!]
+
+    (def mk-key
+      (fn [i]
+        (if (lookup :fancy options)
+          [(pretty-headline i) i]
+          [(pretty-row i) i])))
+
+    (def state-filt
+      (match (map (fn [opt] (read (string-append ":" opt))) (lookup :state options))
+        []      (fn [x] #t)
+        'args   (fn [x] (elem? (lookup :state x) args))))
+
+    (def newness-filt
+      (if (lookup :unread options)
+        (fn [x] (not (lookup :read? x)))
+        (fn [x] #t)))
+
+    (def comment-divider "---")
+
+    (def add-comment-md
+      (fn [x]
+        (string-append
+         "\n" comment-divider "\n"
+         "---Add comment above the line.\n\n"
+         x)))
+
+    (def filtered-items
+      (map mk-key
+           (filter (and-predicate state-filt newness-filt)
+                   (map (fn [i] (enrich-item machine i))
+                        (reverse items)))))
+
+    (def items-dict (dict-from-seq filtered-items))
+    (if (empty-seq? filtered-items)
+      (if (empty-seq? items)
+        (put-str! (string-append "No " (items-name item-flag) "!"))
+        (do
+          (put-str! (string-append "No items of state " (string/intercalate ", " (lookup :state options))  "!"))
+          (put-str! (summary (items-name item-flag) items))))
```

Comment:
This is new behavior. If there are not items at all, the original message is shown. If there are no items of the specified state(s), the specified state(s) and summary is shown.